code/modules/mining/machine_vending.dm

```diff
@@ -55,6 +55,8 @@
 		new /datum/data/mining_equipment("Drone Ranged Upgrade",/obj/item/device/mine_bot_ugprade/cooldown,								600),
 		new /datum/data/mining_equipment("Drone AI Upgrade",	/obj/item/slimepotion/sentience/mining,									1000),
 		new /datum/data/mining_equipment("Jump Boots",			/obj/item/clothing/shoes/bhop,											2500),
+		new /datum/data/mining_equipment("Anti-Ash Walker injector",	/obj/item/weapon/storage/box/scalerot/pen,						2000),
```

Comment:
They are a subspecies of lizard, but their species id is ```ashlizard``` not ```lizard```, hence why your checks fail.